blueflood-core/src/test/java/com/rackspacecloud/blueflood/service/ConfigurationTest.java

```diff
@@ -55,23 +56,170 @@ public void testInitWithBluefloodConfig() throws IOException {
+    }
+
+    @Test
+    public void testTrailingWhitespaceIsKept() {
+
+        String[] expected = { "a", "b", "c   " };
+        List<String> actual = Configuration.stringListFromString("a,b,c   ");
+
+        Assert.assertArrayEquals(expected, actual.toArray());
+    }
+
+    @Test
+    public void testConsecutiveCommasDontProduceEmptyElements() {
+
+        String[] expected = { "a", "b", "c" };
+        List<String> actual = Configuration.stringListFromString("a,,,b,c");
+
+        Assert.assertArrayEquals(expected, actual.toArray());
+    }
+
+    // boolean values
+
+    @Test
+    public void testNullShouldBeInterpretedAsBooleanFalse() {
+
+        Assert.assertFalse(Configuration.booleanFromString(null));
+    }
+
+    @Test
+    public void test_TRUE_ShouldBeInterpretedAsBooleanTrue() {
+
+        Assert.assertTrue(Configuration.booleanFromString("TRUE"));
+    }
+
+    // override behavior
+
     @Test
-    public void testGetListProperty() {
+    public void testSystemPropertiesOverrideConfigurationValues() {
+
+        final String keyName = CoreConfig.MAX_CASSANDRA_CONNECTIONS.toString();
+
         Configuration config = Configuration.getInstance();
-        Assert.assertEquals(config.getStringProperty(CoreConfig.QUERY_MODULES), "");
-        Assert.assertTrue(config.getListProperty(CoreConfig.QUERY_MODULES).isEmpty());
-        System.setProperty("QUERY_MODULES", "a");
-        Assert.assertEquals(config.getListProperty(CoreConfig.QUERY_MODULES).size(), 1);
-        System.setProperty("QUERY_MODULES", "a,b , c");
-        Assert.assertEquals(Arrays.asList("a","b","c"), config.getListProperty(CoreConfig.QUERY_MODULES));
+
+        try {
+            Assert.assertEquals("75",
+                    config.getStringProperty(CoreConfig.MAX_CASSANDRA_CONNECTIONS));
+
+            System.setProperty(keyName, "something else");
+
+            Assert.assertEquals("something else",
+                    config.getStringProperty(CoreConfig.MAX_CASSANDRA_CONNECTIONS));
+        } finally {
+            System.clearProperty(keyName);
+            config.clearProperty(keyName);
+        }
     }
 
+    // originals
+
     @Test
-    public void testBooleanProperty() {
+    public void testGettingValuesCreatesOriginals() {
+
+        final String keyName = CoreConfig.ROLLUP_KEYSPACE.toString();
+
+        try {
+            // arrange
+            Configuration config = Configuration.getInstance();
+
+            Map<Object, Object> props = config.getAllProperties();
+
+            // preconditions
+            Assert.assertTrue(
+                    "props should already have 'ROLLUP_KEYSPACE', because that's a default",
+                    props.containsKey(keyName));
+            Assert.assertFalse(
+                    "props should not contain 'original.ROLLUP_KEYSPACE' yet",
+                    props.containsKey("original." + keyName));
+
+            // act
+            System.setProperty(keyName, "some value");
```

Comment:
Because that is a requirement for the creation of `original.*` key/value pairs. The logic is, admittedly, very convoluted. I'm just trying to capture the behavior of the `getStringProperty` method as it currently works, and will try to clean it up in a subsequent PR.